RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitParametersGPUESProducer.cc

```diff
@@ -0,0 +1,85 @@
+#include <array>
+#include <iostream>
+#include <tuple>
+#include <utility>
+
+#include "EcalRecHitParametersGPURecord.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Framework/interface/ESProductHost.h"
+#include "FWCore/Framework/interface/ESTransientHandle.h"
+#include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/ModuleFactory.h"
+#include "FWCore/Framework/interface/SourceFactory.h"
+#include "FWCore/Framework/interface/eventsetuprecord_registration_macro.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/ReusableObjectHolder.h"
+#include "FWCore/Utilities/interface/typelookup.h"
+#include "RecoLocalCalo/EcalRecAlgos/interface/EcalRecHitParametersGPU.h"
+
+class EcalRecHitParametersGPUESProducer : public edm::ESProducer, public edm::EventSetupRecordIntervalFinder {
+public:
+  EcalRecHitParametersGPUESProducer(edm::ParameterSet const&);
+  ~EcalRecHitParametersGPUESProducer() override = default;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+  std::unique_ptr<EcalRecHitParametersGPU> produce(EcalRecHitParametersGPURecord const&);
+
+protected:
+  void setIntervalFor(const edm::eventsetup::EventSetupRecordKey&,
+                      const edm::IOVSyncValue&,
+                      edm::ValidityInterval&) override;
+
+private:
+  edm::ParameterSet const& pset_;
```

Comment:
The `edm::ParameterSet` argument of the constructor is not guaranteed to live beyond the constructor call, so capturing it to a `const` reference is not safe. I suggest to make a copy instead
```suggestion
  edm::ParameterSet const pset_;
```